docs/ERC721.md

```diff
@@ -0,0 +1,686 @@
+```
+
+### Token Transfers
+
+EIP721 discourages the use of `transferFrom` and favors `safeTransferFrom` in regard to token transfers; where, the safe function adds the following conditional logic:
+1. if the calling address is an externally owned account (EOA), the token transfer will behave as if `transferFrom` was called
+2. if the calling address is a contract, the safe function will check that the contract supports ERC721 tokens
+
+The current implementation of `safeTansferFrom` checks for `onERC721Received`. 
+Because EOAs cannot be differentiated from regular contracts on StarkNet at this time, it's recommended to transfer tokens with `transferFrom`; however, please be aware that transfering tokens to a contract that does not support ERC721 can result in lost tokens forever. 
+
+### ERC721Received
+
+In order to be sure a contract can safely accept ERC721 tokens, said contract must implement the `ERC721_Receiver` interface. Methods such as `safeTransferFrom` and `safeMint` call the recipient contract's `onERC721Received` method. If the contract fails to return the correct magic value, the transaction fails. 
+
+> Please note: EOA/contract differentiation must be resolved in order to correctly use this feature
+
+### Supporting Interfaces
+
+In order to attain seamless layer one interoperability, this ERC721 implementation uses hardcoded interface IDs that have already been calculated on layer one. The Solidity implementation of ERC165 uses the `super` method to amend parent functions. Being that Cairo/Starknet currently does not support this feature, this implementation uses a mapping to store supported interfaces (similar to OpenZeppelin's [ERC165Storage](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/introspection/ERC165Storage.sol)). 
+
+## Extensibility
+
+This implementation of Cairo smart contract extensibility is set up to include all ERC721 storage and function logic within the base contract `ERC721_base.cairo`. Base contract methods with the prefix `ERC721_` must be imported to the frontend contract and inserted into an `external` method with the requisite name. This is already done in `ERC721.cairo`; however, additional functionality can be added. For instance, you could:
+- Implement a pausing mechanism
+- Add roles such as owner or minter
+- Modify the `transferFrom` function to perform actions [before](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC721/ERC721.sol#L335)
+
+Just be sure that the exposed `external` methods invoke their imported function logic a la `approve` invokes `ERC721_approve`. As an example, see below.
+
+```python
+from contracts.token.ERC721_base import ERC721_approve
+
+@external
+func approve{
+        pedersen_ptr: HashBuiltin*, 
+        syscall_ptr: felt*, 
+        range_check_ptr
+    }(to: felt, token_id: Uint256):
+    ERC721_approve(to, token_id)
+    return()
+end
+
+```
+
```

Comment:
I'd add a _Presets_ section to group the "ready to use" frontend contracts (i.e. what's below) with a brief description of the concept and maybe a mention under _Usage_.